### ISSUE TITLE
Unable to autostart redshift via services (Arch Linux)

### ISSUE THREAD

**Issue by mindstormer12** [jonls/redshift#273]
After trying many things, I decided to just start clean and reinstall redshift. Copied the included service to /etc/systemd/system/ and then started and enabled the service. On boot, I get several repeating lines of: 

 >Stopping Redshift display colour temperature adjustment...
 >Starting Redshift display colour temperature adjustment...

`systemctl status redshift.service` shows:

>Failed with result 'exit-code'
>Service hold-off time over, scheduling restart
>Stopped Redshift display colour temperature adjustment
>Start request repeated too quickly

Upon googling, it seems that many people have the same issue and there are several work arounds which I've tried and had no success. 
After trying many things, I decided to just start clean and reinstall redshift. Copied the included service to /etc/systemd/system/ and then started and enabled the service. On boot, I get several repeating lines of: 

 >Stopping Redshift display colour temperature adjustment...
 >Starting Redshift display colour temperature adjustment...

`systemctl status redshift.service` shows:

>Failed with result 'exit-code'
>Service hold-off time over, scheduling restart
>Stopped Redshift display colour temperature adjustment
>Start request repeated too quickly

Upon googling, it seems that many people have the same issue and there are several workarounds which I've tried and had no success.

**Comment by alaviss** [comment]
redshift.service is an user service. It will only work with systemd user-session.

Run

    $ systemd --user start redshift.service

To start redshift for your user (requires Xorg)

**Comment by mindstormer12** [comment]
I actually already tried that--my contents of /etc/systemd/system/redshift.service right now is:

>[Unit]
>Description=Redshift display colour temperature adjustment
>Documentation=http://jonls.dk/redshift/
>After=display-manager.service
>
>[Service]
>Environment=DISPLAY=:0
>ExecStart=/usr/bin/redshift
>Restart=always
>
>[Install]
>WantedBy=default.target

I also have exact same redshift.service at ~/.config/systemd/user/redshift.service as well as a ~/.config/systemd/user/redshift.service.d/display.conf with the following just in case:  

>[Service]
>Environment=DISPLAY=:0

Then ran `systemctl --user enable redshift.service` and `systemctl --user start redshift.service

Do you have redshift auto-started via redshift service? If so, can you post all relevant files and their contents like I did?

Thanks.

**Comment by mindstormer12** [comment]
Apparently the problem exists for `redshift` but not `redshift-gtk`--see [here](https://www.reddit.com/r/archlinux/comments/3wm59x/anyone_got_a_working_redshift_autostart_working/). I'm still not able to solve it.

**Comment by sinapinto** [comment]
Try adding a delay between restarts
```
Restart=always
RestartSec=20
```

**Comment by mindstormer12** [comment]
I added that to the service file and after restarting the computer, redshift.service isn't listed under `systemctl --failed`. However, redshift isn't autostarting and in `systemctl` it says "loaded" under LOAD, "activiating" under ACTIVE, and "auto-restart" after SUB even after waiting a while to give it some time if necessary.

**Comment by Phyks** [comment]
Same here, I have this in my logs:
```
févr. 09 16:55:10 Phyks-laptop systemd[609]: Stopped Redshift display colour temperature adjustment.
févr. 09 16:55:10 Phyks-laptop systemd[609]: Started Redshift display colour temperature adjustment.
févr. 09 16:55:10 Phyks-laptop redshift[697]: `RANDR Query Version' returned error -1
févr. 09 16:55:10 Phyks-laptop redshift[697]: L'initialisation de randr a échouée.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Main process exited, code=exited, status=1/FAILURE
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Unit entered failed state.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Failed with result 'exit-code'.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Service hold-off time over, scheduling restart.
févr. 09 16:55:10 Phyks-laptop systemd[609]: Stopped Redshift display colour temperature adjustment.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Start request repeated too quickly.
févr. 09 16:55:10 Phyks-laptop systemd[609]: Failed to start Redshift display colour temperature adjustment.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Unit entered failed state.
févr. 09 16:55:10 Phyks-laptop systemd[609]: redshift.service: Failed with result 'start-limit'.
```

However, if I start manually the service from a terminal (`systemctl --user start redshift.service`) it works. I think the service may be called too soon, before a graphical server is available.

**Comment by Brottweiler** [comment]
I can start the service just fine but it fails to autostart when enabled;

```sh
● redshift.service - Redshift display colour temperature adjustment
   Loaded: loaded (/usr/lib/systemd/user/redshift.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2017-08-15 12:36:20 CEST; 11s ago
     Docs: http://jonls.dk/redshift/
  Process: 521 ExecStart=/usr/bin/redshift (code=exited, status=1/FAILURE)
 Main PID: 521 (code=exited, status=1/FAILURE)

Aug 15 12:36:19 noname systemd[437]: redshift.service: Unit entered failed state.
Aug 15 12:36:19 noname systemd[437]: redshift.service: Failed with result 'exit-code'.
Aug 15 12:36:20 noname systemd[437]: redshift.service: Service hold-off time over, scheduling restart.
Aug 15 12:36:20 noname systemd[437]: Stopped Redshift display colour temperature adjustment.
Aug 15 12:36:20 noname systemd[437]: redshift.service: Start request repeated too quickly.
Aug 15 12:36:20 noname systemd[437]: Failed to start Redshift display colour temperature adjustment.
Aug 15 12:36:20 noname systemd[437]: redshift.service: Unit entered failed state.
Aug 15 12:36:20 noname systemd[437]: redshift.service: Failed with result 'exit-code'.
```

**Comment by srakitnican** [comment]
Same thing on Fedora. The problem is that Xorg is not started by default display-manager since GDM is running Wayland by default. Then redshift fails to start.

```
Dec 19 11:23:28 machine redshift[1652]: `RANDR Query Version' returned error -1
Dec 19 11:23:28 machine redshift[1652]: Initialization of randr failed.
Dec 19 11:23:28 machine redshift[1652]: Trying next method...
Dec 19 11:23:28 machine redshift[1652]: X request failed: XOpenDisplay
Dec 19 11:23:28 machine redshift[1652]: Initialization of vidmode failed.
Dec 19 11:23:28 machine redshift[1652]: Trying next method...
Dec 19 11:23:28 machine redshift[1652]: No more methods to try.

```

**Comment by Brottweiler** [comment]
Yeah, after I started using [LightDM](https://wiki.archlinux.org/index.php/LightDM) it autostarts, because it waits for LightDM to launch:

```sh
After=display-manager.service
```

**Comment by balalaXD** [comment]
> redshift.service is an user service. It will only work with systemd user-session.
> 
> Run
> 
> ```
> $ systemd --user start redshift.service
> ```
> 
> To start redshift for your user (requires Xorg)

notice without `sudo`

**Comment by sim590** [comment]
I had the same issue with error message

> Dec 19 11:23:28 machine redshift[1652]: `RANDR Query Version' returned error -1
> Dec 19 11:23:28 machine redshift[1652]: Initialization of randr failed.
> Dec 19 11:23:28 machine redshift[1652]: Trying next method...
> Dec 19 11:23:28 machine redshift[1652]: X request failed: XOpenDisplay
> Dec 19 11:23:28 machine redshift[1652]: Initialization of vidmode failed.
> Dec 19 11:23:28 machine redshift[1652]: Trying next method...
> Dec 19 11:23:28 machine redshift[1652]: No more methods to try.

I added the following throught `systemctl --user edit redshift.service`:

```systemd
Environment='XAUTHORITY=/home/MY_USER/.Xauthority' 'DISPLAY=:0'
```

and it now works again...

**Comment by WhyNotHugo** [comment]
Can you check if #753 fixes your issue? I believe it addresses this scenario too.
